src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -58,9 +70,187 @@ def __init__(
+        """
+        starting a timer that runs a given function after a given
+        amount of seconds one time asynchronously.
+
+        :time_in_seconds: the amount of time in seconds
+        :func: the function that is going to be ran
+        """
+        self.timer = QTimer()
+        self.timer.setSingleShot(True)
+        self.timer.timeout.connect(func)
+        self.timer.start(time_in_seconds * MILLISECONDS_PER_SECOND)
+
+    def create_widget(self):
+        """
+        Creating widgets that are used to load, save parameters
+
+        :return: the top of the layout, the bottom of the layout, and edit box widget
+        """
+        save_button = QPushButton("Save Parameters")
+        load_proto_button = QPushButton("Load Parameters")
+        reset_button = QPushButton("Reset All Parameters")
+
+        reset_button.clicked.connect(self.reset_button_callback)
+        save_button.clicked.connect(self.save_proto_callback)
+        load_proto_button.clicked.connect(self.load_proto_with_file_explorer)
+
+        save_hbox_bottom = QHBoxLayout()
+        save_hbox_bottom.addWidget(load_proto_button)
+        save_hbox_bottom.addWidget(reset_button)
+        save_hbox_bottom.addWidget(save_button)
+
+        save_hbox_top = QHBoxLayout()
+        line_box = QLabel(f"Currently Editing: {self.path_to_file}")
+
+        save_hbox_top.addWidget(line_box)
+        return save_hbox_top, save_hbox_bottom, line_box
+
+    def update_proto_from_file(self, path_to_file: str):
+        """
+        load the protobuf from path_to_file to the variable self.proto_to_configure
+
+        :param path_to_file:
+        """
+        if not os.path.isfile(path_to_file):
+            self.proto_to_configure = ThunderbotsConfig()
+            self.proto_to_configure.sensor_fusion_config.friendly_color_yellow = (
+                self.is_yellow
+            )
+
+            self.build_proto(self.proto_to_configure)
+            self.save_current_config_to_file(self.path_to_file)
+            return
+
+        logging.info("Loading protobuf from file: {}".format(path_to_file))
+        with open(path_to_file, "rb") as f:
+            self.proto_to_configure = ThunderbotsConfig()
+            self.proto_to_configure.ParseFromString(f.read())
+
+            self.proto_to_configure.sensor_fusion_config.friendly_color_yellow = (
+                self.is_yellow
+            )
+
+    def send_proto_to_fullsystem(self):
+        """
+        Sending the default configuration protobufs to unix_full_system at startup.
+        Also updates the widget at the same time.
+        """
+        self.update_proto_from_file(self.path_to_file)
+        self.on_change_callback(None, None, self.proto_to_configure)
+        self.update_widget()
+
+    def save_proto_callback(self):
+        """
+        This is a callback for a button that save the current protobuf to disk!
+        """
+        self.save_current_config_to_file(self.path_to_file)
```

Comment:
We should also have a file dialogue box for saving so we can save multiple different parameters settings. Though I think it'd be useful to then also update the Widget to load the latest saved protobuf on launch. Though maybe we still need this so we can set the `sensor_fusion_config.friendly_color_yellow` parameter depending on the color. @itsarune might know better about that.

Reply:
Can you clarify the question? I think the way vincent implemented it makes sense, where `is_yellow` gets manually set in `sensor_fusion_config.friendly_colour_yellow`

Reply:
we still need to know whether or not the parameter is blue or yellow because of sensor fusion. Also, this field would be overwritten based on the parameter that are passed into `ProtoConfigurationWidget` class